### PR TITLE
feat: implement MFA in taskmaster

### PR DESCRIPTION
**Pull request by Crunchyman-ralph**
# What type of PR is this?

 - [ ] 🐛 Bug fix
 - [ ] ✨ Feature
 - [ ] 🔌 Integration
 - [ ] 📝 Docs
 - [ ] 🧹 Refactor
 - [ ] Other:
## How to Test This

```bash
# Example commands or steps
```

**Expected result:**

## Contributor Checklist

- [ ] Created changeset: `npm run changeset`
- [ ] Tests pass: `npm test`
- [ ] Format check passes: `npm run format-check` (or `npm run format` to fix)
- [ ] Addressed CodeRabbit comments (if any)
- [ ] Linked related issues (if any)
- [ ] Manually tested the changes

## Changelog Entry

---

### For Maintainers

- [ ] PR title follows conventional commits
- [ ] Target branch correct
- [ ] Labels added
- [ ] Milestone assigned (if applicable)

## Summary by CodeRabbit

* **New Features**
  * MFA support across CLI and auth services: users may be prompted for a 6‑digit MFA code during login or token flows; successful MFA returns credentials. New centralized session manager handles session/token lifecycle and improved local sign-out behavior.

* **Bug Fixes**
  * Centralized, MFA-aware error handling with clearer prompts, cancellation support, retry (configurable, default up to 3 attempts) and consistent messaging; help text notes MFA cannot be skipped.

* **Tests**
  * Expanded coverage for session management, MFA flows, retry logic, and edge cases.

* **Documentation**
  * Expanded testing and architecture guidelines.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: apps/cli/src/commands/auth.command.ts [hunk 1]
```diff
@@ -504,19 +517,131 @@ Examples:
...
+
+				// MFA is required - prompt the user for their MFA code
+				return this.handleMFAVerification(error);
+			}
+
+			// Only show "Authentication failed" for actual failures
 			spinner.fail('Authentication failed');
 			throw error;
 		}
 	}
```
[the remark this answers is not in the document]
Comment (Crunchyman-ralph) [reply apps/cli/src/commands/auth.command.ts]: for point 1, we can't actually bypass this since we need the mfa code to actually interact with the app.

If mfa is not there, we won't actually be able to use the api-storage. so even in --yes, we still need user input for mfa, this is the only exception.
